CODE_GUIDE.md

```diff
@@ -0,0 +1,56 @@
+# Code Standards and Style Guide
+
+## Format
+
+Our base code style is [PEP8](https://www.python.org/dev/peps/pep-0008/), but we allow a max line length of 120.
+
+Check formatting and dead code with
+
+```shell
+make lint
+```
+
+## Code Standards
+
+### Context Use
+
+We make use of the [Behave Context](https://behave.readthedocs.io/en/stable/tutorial.html#context) object to store data
+that is needed across multiple steps of a scenario.
+
+To prevent the context from becoming cluttered and confusing to use we define some rules for how we interact with it:
+
+- Only step functions and environment hooks should interact with the context attributes
+- Context attributes should be documented in the [Context Index](#Context Index)
+
```

Comment:
I'd like to flesh this out a bit something like:

Other none @step functions shouldn't be passed context and certainly not modify it.  
Instead pass in explicit variables from the context and return new ones as required.

e.g
step
def send_a_message_for_receipt(context):
    context.some_var =  helper_function(context.emitted_cases_id[0])




 be pure/deterministic functions